server/controllers/finance/reports/ohada_balance_sheet/index.js

```diff
@@ -0,0 +1,299 @@
+          _.extend(item, aggregateReferences(list, currentReferences, previousReferences));
+        }
+        return item;
+      });
+
+      const liabilityTable = balanceSheetLiabilityTable.map(item => {
+        item.label = 'REPORT.OHADA.REF_DESCRIPTION.'.concat(item.ref);
+        if (currentReferences[item.ref]) {
+          item.currentBrut = currentReferences[item.ref].brut.balance;
+          item.currentAmo = currentReferences[item.ref].amortissement.balance;
+          item.currentNet = currentReferences[item.ref].net.balance;
+          item.previousNet = previousReferences[item.ref] ? previousReferences[item.ref].net.balance : 0;
+        }
+
+        // process manually totals
+        if (item.ref === 'CP') {
+          list = ['CL', 'CM'];
+          _.extend(item, aggregateReferences(list, currentReferences, previousReferences));
+        }
+
+        if (item.ref === 'DF') {
+          list = ['DA', 'DB', 'DC', 'DD'];
+          _.extend(item, aggregateReferences(list, currentReferences, previousReferences));
+        }
+
+        if (item.ref === 'DG') {
+          list = ['CL', 'CM', 'DA', 'DB', 'DC', 'DD'];
+          _.extend(item, aggregateReferences(list, currentReferences, previousReferences));
+        }
+
+        if (item.ref === 'DP') {
+          list = ['DH', 'DI', 'DJ', 'DK', 'DL', 'DM', 'DN'];
+          _.extend(item, aggregateReferences(list, currentReferences, previousReferences));
+        }
+
+        if (item.ref === 'DT') {
+          list = ['DQ', 'DR', 'DS'];
+          _.extend(item, aggregateReferences(list, currentReferences, previousReferences));
+        }
+
+        if (item.ref === 'DZ') {
+          list = ['CL', 'CM', 'DA', 'DB', 'DC', 'DD']
+            .concat(['DH', 'DI', 'DJ', 'DK', 'DL', 'DM', 'DN'])
+            .concat(['DQ', 'DR', 'DS'])
+            .concat(['DV']);
+          _.extend(item, aggregateReferences(list, currentReferences, previousReferences));
+        }
+        return item;
+      });
+
+      _.merge(context, { assetTable, liabilityTable });
+      return report.render(context);
+    })
+    .then(result => {
+      res.set(result.header).send(result.report);
+    })
+    .catch(next);
+}
+
+function aggregateReferences(references, currentDb, previousDb) {
+  const item = {
+    currentBrut : 0, currentAmo : 0, currentNet : 0, previousNet : 0,
+  };
+
+  references.forEach(ref => {
+    item.currentBrut += currentDb[ref] ? currentDb[ref].brut.balance : 0;
+    item.currentAmo += currentDb[ref] ? currentDb[ref].amortissement.balance : 0;
+    item.currentNet += currentDb[ref] ? currentDb[ref].net.balance : 0;
+    item.previousNet += previousDb[ref] ? previousDb[ref].net.balance : 0;
+  });
+
+  return item;
+}
+
+function formatReferences(references) {
+  const values = {};
+  Object.keys(references).forEach(key => {
+    const [brut] = references[key].filter(elt => elt.is_amo_dep === 0);
+    let [amortissement] = references[key].filter(elt => elt.is_amo_dep === 1);
```

Comment:
`is_amo_dep` must be either 0 or 1 always, right?  Can we use [`_.partition()`](https://lodash.com/docs/4.17.10#partition) for this?
```js
let [brut, amortissement] = _.partition(references[key], elt => elt.is_amo_dep === 0);
```

Reply:
Wait... what are the types of `brut` and `amortissment`?  Are they always objects?  Or can they be arrays?